bftengine/src/bftengine/KeyManager.cpp

```diff
@@ -0,0 +1,96 @@
+#include "KeyManager.h"
+#include "thread"
+#include "ReplicaImp.hpp"
+#include "ReplicaConfig.hpp"
+
+const std::string KeyManager::KeyExchangeMsg::getVersion() const { return "1"; }
+
+KeyManager::KeyExchangeMsg KeyManager::KeyExchangeMsg::MsgDeserializer(const char* serializedMsg,
+                                                                       const uint32_t& size) {
+  std::stringstream ss;
+  KeyManager::KeyExchangeMsg ke;
+  ss.write(serializedMsg, std::streamsize(size));
+  KeyManager::KeyExchangeMsg::deserialize(ss, ke);
+  return ke;
+}
+
+void KeyManager::KeyExchangeMsg::serializeDataMembers(std::ostream& outStream) const {
+  serialize(outStream, key_);
+  LOG_TRACE(GL, "KEY EXCHANGE MANAGER  ser  key_ " << key_);
+  serialize(outStream, signature_);
+  LOG_TRACE(GL, "KEY EXCHANGE MANAGER  ser  signature_ " << signature_);
+  serialize(outStream, repID_);
+  LOG_TRACE(GL, "KEY EXCHANGE MANAGER  ser  repID_ " << repID_);
+}
+
+void KeyManager::KeyExchangeMsg::deserializeDataMembers(std::istream& inStream) {
+  deserialize(inStream, key_);
+  deserialize(inStream, signature_);
+  deserialize(inStream, repID_);
+}
+
+std::string KeyManager::KeyExchangeMsg::toString() const {
+  std::stringstream ss;
+  ss << "key [" << key_ << "] signature [" << signature_ << "] replica id [" << repID_ << "]";
+  return ss.str();
+}
+
+KeyManager& KeyManager::get() {
+  static KeyManager km;
+  return km;
+}
+void KeyManager::set(concord::kvbc::IClient* cl, const int& id) {
+  cl_ = cl;
+  repID_ = id;
+}
+
+// Method for testing periodic generation
```

Comment:
I really don't like the idea of submitting commented code.
The fact that the feature has a feature flag is not an excuse.
Please remove or move to unit tests.